etc/picongpu/taurus-tud/knl_picongpu.profile.example

```diff
@@ -0,0 +1,45 @@
+# Text Editor for Tools #######################################################
+#   - examples: "nano", "vim", "emacs -nw", "vi" or without terminal: "gedit"
+#export EDITOR="nano"
+
+# "module" commands undefine $BASH_SOURCE on a KNL node for some unknown reason
+# so $PIC_PROFILE needs to be defined before using "module ...".
+export PIC_PROFILE=$(cd $(dirname $BASH_SOURCE) && pwd)"/"$(basename $BASH_SOURCE)
+
+# General modules #############################################################
+#
+module purge
+module load cmake/3.9.0
+module load git
+module load intelmpi/2017.2.174
+
+# Compilers ###################################################################
+### ICC
+module load intel/2017.2.174
+# Boost needs to be build yourself for now!
+# The boost root path needs to be adjusted.
+export BOOST_ROOT=<BOOST_INSTALLATION_DIR>
+export LD_LIBRARY_PATH=$BOOST_ROOT:$LD_LIBRARY_PATH
+export LD_LIBRARY_PATH=$BOOST_ROOT/lib:$LD_LIBRARY_PATH
+
+export PICSRC=$HOME/src/picongpu
+export PIC_BACKEND="omp2b:native"
```

Comment:
if a user compiles an the head node, we should better make this `omp2b:MIC-AVX512`

```
icc --help codegen

Code Generation
---------------

-x<code>  generate specialized code to run exclusively on processors
          indicated by <code> as described below
            SSE2    May generate Intel(R) SSE2 and SSE instructions for Intel
                    processors.  Optimizes for the Intel NetBurst(R)
                    microarchitecture.
            SSE3    May generate Intel(R) SSE3, SSE2, and SSE instructions for
                    Intel processors.  Optimizes for the enhanced Pentium(R) M 
                    processor microarchitecture and Intel NetBurst(R)
                    microarchitecture. 
            SSSE3   May generate Intel(R) SSSE3, SSE3, SSE2, and SSE
                    instructions for Intel processors.  Optimizes for the
                    Intel(R) Core(TM) microarchitecture.
            SSE4.1  May generate Intel(R) SSE4 Vectorizing Compiler and Media
                    Accelerator instructions for Intel processors.  May 
                    generate Intel(R) SSSE3, SSE3, SSE2, and SSE instructions
                    and it may optimize for Intel(R) 45nm Hi-k next generation
                    Intel Core(TM) microarchitecture.
            SSE4.2  May generate Intel(R) SSE4 Efficient Accelerated String
                    and Text Processing instructions supported by Intel(R)
                    Core(TM) i7 processors.  May generate Intel(R) SSE4 
                    Vectorizing Compiler and Media Accelerator, Intel(R) SSSE3,
                    SSE3, SSE2, and SSE instructions and it may optimize for
                    the Intel(R) Core(TM) processor family.
            AVX     May generate Intel(R) Advanced Vector Extensions (Intel(R)
                    AVX), Intel(R) SSE4.2, SSE4.1, SSSE3, SSE3,
                    SSE2, and SSE instructions for Intel(R) processors.
            CORE-AVX2
                    May generate Intel(R) Advanced Vector Extensions 2
                    (Intel(R) AVX2), Intel(R) AVX, SSE4.2, SSE4.1, SSSE3, SSE3,
                    SSE2, and SSE instructions for Intel(R) processors.
            CORE-AVX-I
                    May generate Intel(R) Advanced Vector Extensions (Intel(R)
                    AVX), including instructions in Intel(R) Core 2(TM)
                    processors in process technology smaller than 32nm,
                    Intel(R) SSE4.2, SSE4.1, SSSE3, SSE3, SSE2, and SSE
                    instructions for Intel(R) processors.
            ATOM_SSE4.2
                    May generate MOVBE instructions for Intel(R) processors,
                    depending on the setting of option -minstruction.
                    May also generate Intel(R) SSE4.2, SSE3, SSE2, and SSE
                    instructions for Intel processors. Optimizes for Intel(R)
                    Atom(TM) processors that support Intel(R) SSE4.2 and MOVBE
                    instructions.
            ATOM_SSSE3
                    May generate MOVBE instructions for Intel(R) processors,
                    depending on the setting of option -minstruction.
                    May also generate Intel(R) SSSE3, SSE3, SSE2, and SSE
                    instructions for Intel processors. Optimizes for the
                    Intel(R) Atom(TM) processor that support Intel(R) SSE
                    and MOVBE instructions.
            MIC-AVX512
                    May generate Intel(R) Advanced Vector Extensions 512
                    (Intel(R) AVX-512) Foundation instructions, Intel(R)
                    AVX-512 Conflict Detection instructions, Intel(R) AVX-512
                    Exponential and Reciprocal instructions, Intel(R) AVX-512
                    Prefetch instructions for Intel(R) processors, and the
                    instructions enabled with CORE-AVX2. Optimizes for Intel(R)
                    processors that support Intel(R) AVX-512 instructions.
            CORE-AVX512 
                    May generate Intel(R) Advanced Vector Extensions 512 
                    (Intel(R) AVX-512) Foundation instructions, Intel(R) 
                    AVX-512 Conflict Detection instructions, Intel(R) AVX-512 
                    Doubleword and Quadword instructions, Intel(R) AVX-512 
                    Byte and Word instructions and Intel(R) AVX-512 Vector 
                    Length Extensions for Intel(R) processors, and the 
                    instructions enabled with CORE-AVX2. Optimizes for Intel(R)
                    processors that support Intel(R) AVX-512 instructions. 
            COMMON-AVX512
               	    May generate Intel(R) Advanced Vector Extensions 512
                    (Intel(R) AVX-512) Foundation instructions, Intel(R)
                    AVX-512 Conflict Detection instructions, as well as the
                    instructions enabled with CORE-AVX2. Optimizes for Intel(R)
                    processors that support Intel(R) AVX-512 instructions.
```

Reply:
You can't compile for the KNL on Taurus on the head node, because they were so smart to install a more recent version of CentOS on the KNLs than on the head node **and every other node**, which also breaks **central, node wide** the module systems e.g. for some modules like boost.

Reply:
*slow clap*

Reply:
It's not like this would be super handy as compiling PIConGPU with 1,3Ghz single core performance is no fun. :unamused:

Reply:
Ah, yet an other use case that would not be a problem with [spack](http://spack.readthedocs.io/).

Reply:
But seriously, open a ticket about cross-compiling on the head node?

Reply:
And still add `:MIC-AVX512` just to be explicit?

Reply:
ax3l, the people from Taurus know about this problem but as all nodes will be updated soon™ so CentOS 7, they will not fix this for now.

Reply:
> And still add :MIC-AVX512 just to be explicit?

Reply:
If you insist...